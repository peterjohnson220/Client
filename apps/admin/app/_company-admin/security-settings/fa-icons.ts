
import { faPlus } from '@fortawesome/pro-solid-svg-icons/faPlus';
import { faLongArrowAltLeft } from '@fortawesome/pro-solid-svg-icons/faLongArrowAltLeft';
import { faPencil } from '@fortawesome/pro-solid-svg-icons/faPencil';
import { faCheckCircle } from '@fortawesome/pro-solid-svg-icons/faCheckCircle';
import { faTimes } from '@fortawesome/pro-solid-svg-icons/faTimes';
import { faTrash } from '@fortawesome/pro-solid-svg-icons/faTrash';
import { faSpinner } from '@fortawesome/pro-solid-svg-icons/faSpinner';

export const faIcons = [
  faPlus, faLongArrowAltLeft, faPencil, faCheckCircle, faTimes, faTrash, faSpinner
];
