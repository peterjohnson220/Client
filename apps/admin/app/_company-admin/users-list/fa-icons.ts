import { faCircle } from '@fortawesome/pro-solid-svg-icons/faCircle';
import { faMinusCircle } from '@fortawesome/pro-solid-svg-icons/faMinusCircle';
import { faUser } from '@fortawesome/pro-solid-svg-icons/faUser';

export const faIcons = [faCircle, faMinusCircle, faUser];
