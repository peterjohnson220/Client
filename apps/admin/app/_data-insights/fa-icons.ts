import { faSync } from '@fortawesome/pro-solid-svg-icons/faSync';

export const faIcons = [
  faSync
];
