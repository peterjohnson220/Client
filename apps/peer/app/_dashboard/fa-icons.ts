import { faBuilding } from '@fortawesome/pro-solid-svg-icons/faBuilding';
import { faBriefcase } from '@fortawesome/pro-solid-svg-icons/faBriefcase';
import { faTimes } from '@fortawesome/pro-regular-svg-icons/faTimes';
import { faInfoCircle } from '@fortawesome/pro-regular-svg-icons/faInfoCircle';
import { faUsdCircle } from '@fortawesome/pro-solid-svg-icons/faUsdCircle';
import { faUsers } from '@fortawesome/pro-solid-svg-icons/faUsers';
import { faCaretDown } from '@fortawesome/pro-solid-svg-icons/faCaretDown';

export const faIcons = [faBuilding, faBriefcase, faTimes, faInfoCircle, faUsdCircle, faUsers, faCaretDown];
