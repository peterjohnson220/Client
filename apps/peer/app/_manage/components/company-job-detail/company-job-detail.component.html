<h5 class="text-uppercase mb-3">Company Job</h5>
<div>
  <label class="text-muted text-uppercase">Title</label>
  <span>{{ selectedCompanyJob.JobTitle }}</span>
</div>
<div>
  <label class="text-muted text-uppercase">Family</label>
  <span>{{ selectedCompanyJob.JobFamily }}</span>
</div>
<div>
  <label class="text-muted text-uppercase">Level</label>
  <span>{{ selectedCompanyJob.JobLevel }}</span>
</div>
<div>
  <label class="text-muted text-uppercase">Description</label>
  <a href="javascript:void(0)" [class.show]="jdmDescriptionIds.length === 1" class="jdm mr-2" (click)="handleViewJdmDescriptionClick()">View Full Job Description</a>
  <p class="mb-1" *ngIf="jdmDescriptionIds.length > 1">
    There is more than one Job Description for this job. Please refer to the 
      <a href="/ng/job-description-management/job-descriptions" target="_blank">Job Description Management Tile</a> 
    to view each description.
  </p>
  <pf-loading-indicator *ngIf="jdmDescriptionLoading"></pf-loading-indicator>
  <p *ngIf="jdmDescriptionLoadingError" class="jdm-error d-inline-block mb-0">
    Unable to load job description
  </p>
  <ng-container *ngIf="selectedCompanyJob.JobDescription && selectedCompanyJob.JobDescription !== '#N/A'; else noDescription">
    <p [innerHTML]="selectedCompanyJob.JobDescription | jobDescriptionParser" class="job-description"></p>
  </ng-container>
  <ng-template #noDescription>
    <p *ngIf="!jdmDescriptionIds.length > 0" class="font-italic">
      No Job Description Available
    </p>
  </ng-template>
</div>