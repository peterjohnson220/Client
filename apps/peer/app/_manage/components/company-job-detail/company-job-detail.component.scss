@import 'colors';

label {
  min-width: 90px;
}

a.jdm {
  transition: opacity .3s ease-in-out;
  &.d-none {
    display:none;
  }
  &:hover {
    color: $link-color;
  }
}

.jdm-error {
  color: tomato;
}

.company-job-description-container {
  flex:1;
  overflow-y:auto;
}
