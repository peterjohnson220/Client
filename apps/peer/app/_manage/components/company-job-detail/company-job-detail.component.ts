import { Component, Input, Output, EventEmitter } from '@angular/core';

import { CompanyJob } from 'libs/models/company';

@Component({
  selector: 'pf-company-job-detail',
  templateUrl: './company-job-detail.component.html',
  styleUrls: ['./company-job-detail.component.scss']
})
export class CompanyJobDetailComponent {
  @Input() selectedCompanyJob: CompanyJob;
  @Input() jdmDescriptionIds: number[];
  @Input() jdmDescriptionLoading: boolean;
  @Input() jdmDescriptionLoadingError: boolean;

  @Output() viewJdmDescriptionClick = new EventEmitter();

  jdmUrl = '/client/job-description-management/job-descriptions';

  handleViewJdmDescriptionClick() {
    this.viewJdmDescriptionClick.emit();
  }
}
