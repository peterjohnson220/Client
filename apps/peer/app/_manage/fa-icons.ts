import { faExclamationTriangle } from '@fortawesome/pro-solid-svg-icons/faExclamationTriangle';
import { faDownload } from '@fortawesome/pro-solid-svg-icons/faDownload';
import { faUpload } from '@fortawesome/pro-solid-svg-icons/faUpload';
import { faInfoCircle } from '@fortawesome/pro-solid-svg-icons';
import { faSitemap } from '@fortawesome/pro-solid-svg-icons/faSitemap';
import { faTrashAlt } from '@fortawesome/pro-solid-svg-icons/faTrashAlt';
import { faChevronCircleLeft } from '@fortawesome/pro-solid-svg-icons/faChevronCircleLeft';
import { faChevronCircleRight } from '@fortawesome/pro-solid-svg-icons/faChevronCircleRight';
import { faBriefcase } from '@fortawesome/pro-solid-svg-icons/faBriefcase';
import { faRectangleLandscape } from '@fortawesome/pro-solid-svg-icons/faRectangleLandscape';
import { faBalanceScaleRight } from '@fortawesome/pro-solid-svg-icons/faBalanceScaleRight';
import { faTimes } from '@fortawesome/pro-solid-svg-icons/faTimes';
import { faCircle } from '@fortawesome/pro-solid-svg-icons/faCircle';
import { faExchange } from '@fortawesome/pro-solid-svg-icons/faExchange';
import { faRedoAlt } from '@fortawesome/pro-solid-svg-icons/faRedoAlt';

export const faIcons = [faExclamationTriangle, faDownload, faUpload, faInfoCircle, faSitemap, faTrashAlt,
  faChevronCircleLeft, faChevronCircleRight, faBriefcase, faRectangleLandscape, faBalanceScaleRight, faTimes,
  faCircle, faExchange, faRedoAlt];
