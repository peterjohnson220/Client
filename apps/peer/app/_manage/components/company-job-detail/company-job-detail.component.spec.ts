import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { NO_ERRORS_SCHEMA } from '@angular/core';

import { generateMockCompanyJob } from 'libs/models/company/company-job.model';
import { JobDescriptionParserPipe } from 'libs/core/pipes';
import { FontAwesomeModule } from '@fortawesome/angular-fontawesome';
import * as fromFaIcons from 'libs/core/fa-icons';

import { CompanyJobDetailComponent } from './company-job-detail.component';

describe('CompanyJobDetailComponent', () => {
  let component: CompanyJobDetailComponent;
  let fixture: ComponentFixture<CompanyJobDetailComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [CompanyJobDetailComponent, JobDescriptionParserPipe],
      imports: [FontAwesomeModule],
      // Shallow Testing
      schemas: [NO_ERRORS_SCHEMA]
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(CompanyJobDetailComponent);
    component = fixture.componentInstance;
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should render the selected company jobs props', () => {
    component.jdmDescriptionIds = [];
    component.selectedCompanyJob = generateMockCompanyJob();

    // set JobDescription to blank to avoid the parser from executing and throwing errors
    component.selectedCompanyJob.JobDescription = '';

    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should show a link to download the jdm description via the `show` class when one jdm description id exists', () => {
    component.jdmDescriptionIds = [123456];
    component.selectedCompanyJob = generateMockCompanyJob();

    // set JobDescription to blank to avoid the parser from executing and throwing errors
    component.selectedCompanyJob.JobDescription = '';

    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should hide a link to download the jdm description via omitting the `show` class when no jdm descriptions exists', () => {
    component.jdmDescriptionIds = [];
    component.selectedCompanyJob = generateMockCompanyJob();

    // set JobDescription to blank to avoid the parser from executing and throwing errors
    component.selectedCompanyJob.JobDescription = '';

    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should show an informational message when multiple jdm ids exist', () => {
    component.jdmDescriptionIds = [123, 456];
    component.selectedCompanyJob = generateMockCompanyJob();

    // set JobDescription to blank to avoid the parser from executing and throwing errors
    component.selectedCompanyJob.JobDescription = '';

    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should show a spinner when downloading a jdm description', () => {
    component.jdmDescriptionIds = [123];
    component.jdmDescriptionLoading = true;
    component.selectedCompanyJob = generateMockCompanyJob();

    // set JobDescription to blank to avoid the parser from executing and throwing errors
    component.selectedCompanyJob.JobDescription = '';

    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should show an error message if the jdm description download fails', () => {
    component.jdmDescriptionIds = [123];
    component.jdmDescriptionLoading = false;
    component.jdmDescriptionLoadingError = true;
    component.selectedCompanyJob = generateMockCompanyJob();

    // set JobDescription to blank to avoid the parser from executing and throwing errors
    component.selectedCompanyJob.JobDescription = '';

    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should emit when the view full job description link is clicked', () => {
    spyOn(component.viewJdmDescriptionClick, 'emit');

    component.jdmDescriptionIds = [123];
    component.jdmDescriptionLoading = false;
    component.jdmDescriptionLoadingError = false;
    component.selectedCompanyJob = generateMockCompanyJob();

    // set JobDescription to blank to avoid the parser from executing and throwing errors
    component.selectedCompanyJob.JobDescription = '';

    fixture.detectChanges();

    const viewFullJobDescription = fixture.debugElement.nativeElement.querySelector('a.jdm');
    viewFullJobDescription.click();

    expect(component.viewJdmDescriptionClick.emit).toHaveBeenCalled();
  });
});
