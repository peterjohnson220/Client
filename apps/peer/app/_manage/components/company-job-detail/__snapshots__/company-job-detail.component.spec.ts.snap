// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompanyJobDetailComponent should hide a link to download the jdm description via omitting the \`show\` class when no jdm descriptions exists 1`] = `
<pf-company-job-detail
  enableCoreJdmInClient$={[Function Store]}
  enableCoreJdmInClientSubscription={[Function Subscriber]}
  jdmDescriptionIds={[Function Array]}
  jdmUrl={[Function String]}
  selectedCompanyJob={[Function Object]}
  settingService={[Function SettingsService]}
  viewJdmDescriptionClick={[Function EventEmitter]}
>
  <h5
    class="mb-3 font-weight-bold"
  >
    Company Job
  </h5><div
    class="info-row"
  >
    <label
      class="font-weight-bold w-25 mb-0"
    >
      Title:
    </label>
    <span>
      MockCompanyJobTitle
    </span>
  </div><div
    class="info-row"
  >
    <label
      class="font-weight-bold w-25 mb-0"
    >
      Family:
    </label>
    <span>
      MockJobFamily
    </span>
  </div><div
    class="info-row"
  >
    <label
      class="font-weight-bold w-25 mb-0"
    >
      Level:
    </label>
    <span>
      MockJobLevel
    </span>
  </div><div
    class="info-row border-0"
  >
    <label
      class="font-weight-bold w-25"
    >
      Description:
    </label>
    <a
      class="jdm mr-2"
      href="javascript:void(0)"
    >
      View Full Job Description
    </a>
    
    
    
    
    
    <p
      class="font-italic"
    >
       No Job Description Available 
    </p>
    
    
  </div>
</pf-company-job-detail>
`;

exports[`CompanyJobDetailComponent should render the selected company jobs props 1`] = `
<pf-company-job-detail
  enableCoreJdmInClient$={[Function Store]}
  enableCoreJdmInClientSubscription={[Function Subscriber]}
  jdmDescriptionIds={[Function Array]}
  jdmUrl={[Function String]}
  selectedCompanyJob={[Function Object]}
  settingService={[Function SettingsService]}
  viewJdmDescriptionClick={[Function EventEmitter]}
>
  <h5
    class="mb-3 font-weight-bold"
  >
    Company Job
  </h5><div
    class="info-row"
  >
    <label
      class="font-weight-bold w-25 mb-0"
    >
      Title:
    </label>
    <span>
      MockCompanyJobTitle
    </span>
  </div><div
    class="info-row"
  >
    <label
      class="font-weight-bold w-25 mb-0"
    >
      Family:
    </label>
    <span>
      MockJobFamily
    </span>
  </div><div
    class="info-row"
  >
    <label
      class="font-weight-bold w-25 mb-0"
    >
      Level:
    </label>
    <span>
      MockJobLevel
    </span>
  </div><div
    class="info-row border-0"
  >
    <label
      class="font-weight-bold w-25"
    >
      Description:
    </label>
    <a
      class="jdm mr-2"
      href="javascript:void(0)"
    >
      View Full Job Description
    </a>
    
    
    
    
    
    <p
      class="font-italic"
    >
       No Job Description Available 
    </p>
    
    
  </div>
</pf-company-job-detail>
`;

exports[`CompanyJobDetailComponent should show a link to download the jdm description via the \`show\` class when one jdm description id exists 1`] = `
<pf-company-job-detail
  enableCoreJdmInClient$={[Function Store]}
  enableCoreJdmInClientSubscription={[Function Subscriber]}
  jdmDescriptionIds={[Function Array]}
  jdmUrl={[Function String]}
  selectedCompanyJob={[Function Object]}
  settingService={[Function SettingsService]}
  viewJdmDescriptionClick={[Function EventEmitter]}
>
  <h5
    class="mb-3 font-weight-bold"
  >
    Company Job
  </h5><div
    class="info-row"
  >
    <label
      class="font-weight-bold w-25 mb-0"
    >
      Title:
    </label>
    <span>
      MockCompanyJobTitle
    </span>
  </div><div
    class="info-row"
  >
    <label
      class="font-weight-bold w-25 mb-0"
    >
      Family:
    </label>
    <span>
      MockJobFamily
    </span>
  </div><div
    class="info-row"
  >
    <label
      class="font-weight-bold w-25 mb-0"
    >
      Level:
    </label>
    <span>
      MockJobLevel
    </span>
  </div><div
    class="info-row border-0"
  >
    <label
      class="font-weight-bold w-25"
    >
      Description:
    </label>
    <a
      class="jdm mr-2 show"
      href="javascript:void(0)"
    >
      View Full Job Description
    </a>
    
    
    
    
    
    
    
  </div>
</pf-company-job-detail>
`;

exports[`CompanyJobDetailComponent should show a spinner when downloading a jdm description 1`] = `
<pf-company-job-detail
  enableCoreJdmInClient$={[Function Store]}
  enableCoreJdmInClientSubscription={[Function Subscriber]}
  jdmDescriptionIds={[Function Array]}
  jdmDescriptionLoading={[Function Boolean]}
  jdmUrl={[Function String]}
  selectedCompanyJob={[Function Object]}
  settingService={[Function SettingsService]}
  viewJdmDescriptionClick={[Function EventEmitter]}
>
  <h5
    class="mb-3 font-weight-bold"
  >
    Company Job
  </h5><div
    class="info-row"
  >
    <label
      class="font-weight-bold w-25 mb-0"
    >
      Title:
    </label>
    <span>
      MockCompanyJobTitle
    </span>
  </div><div
    class="info-row"
  >
    <label
      class="font-weight-bold w-25 mb-0"
    >
      Family:
    </label>
    <span>
      MockJobFamily
    </span>
  </div><div
    class="info-row"
  >
    <label
      class="font-weight-bold w-25 mb-0"
    >
      Level:
    </label>
    <span>
      MockJobLevel
    </span>
  </div><div
    class="info-row border-0"
  >
    <label
      class="font-weight-bold w-25"
    >
      Description:
    </label>
    <a
      class="jdm mr-2 show"
      href="javascript:void(0)"
    >
      View Full Job Description
    </a>
    
    
    <pf-loading-indicator />
    
    
    
    
    
  </div>
</pf-company-job-detail>
`;

exports[`CompanyJobDetailComponent should show an error message if the jdm description download fails 1`] = `
<pf-company-job-detail
  enableCoreJdmInClient$={[Function Store]}
  enableCoreJdmInClientSubscription={[Function Subscriber]}
  jdmDescriptionIds={[Function Array]}
  jdmDescriptionLoading="false"
  jdmDescriptionLoadingError={[Function Boolean]}
  jdmUrl={[Function String]}
  selectedCompanyJob={[Function Object]}
  settingService={[Function SettingsService]}
  viewJdmDescriptionClick={[Function EventEmitter]}
>
  <h5
    class="mb-3 font-weight-bold"
  >
    Company Job
  </h5><div
    class="info-row"
  >
    <label
      class="font-weight-bold w-25 mb-0"
    >
      Title:
    </label>
    <span>
      MockCompanyJobTitle
    </span>
  </div><div
    class="info-row"
  >
    <label
      class="font-weight-bold w-25 mb-0"
    >
      Family:
    </label>
    <span>
      MockJobFamily
    </span>
  </div><div
    class="info-row"
  >
    <label
      class="font-weight-bold w-25 mb-0"
    >
      Level:
    </label>
    <span>
      MockJobLevel
    </span>
  </div><div
    class="info-row border-0"
  >
    <label
      class="font-weight-bold w-25"
    >
      Description:
    </label>
    <a
      class="jdm mr-2 show"
      href="javascript:void(0)"
    >
      View Full Job Description
    </a>
    
    
    
    <p
      class="jdm-error d-inline-block mb-0"
    >
       Unable to load job description 
    </p>
    
    
    
    
  </div>
</pf-company-job-detail>
`;

exports[`CompanyJobDetailComponent should show an informational message when multiple jdm ids exist 1`] = `
<pf-company-job-detail
  enableCoreJdmInClient$={[Function Store]}
  enableCoreJdmInClientSubscription={[Function Subscriber]}
  jdmDescriptionIds={[Function Array]}
  jdmUrl={[Function String]}
  selectedCompanyJob={[Function Object]}
  settingService={[Function SettingsService]}
  viewJdmDescriptionClick={[Function EventEmitter]}
>
  <h5
    class="mb-3 font-weight-bold"
  >
    Company Job
  </h5><div
    class="info-row"
  >
    <label
      class="font-weight-bold w-25 mb-0"
    >
      Title:
    </label>
    <span>
      MockCompanyJobTitle
    </span>
  </div><div
    class="info-row"
  >
    <label
      class="font-weight-bold w-25 mb-0"
    >
      Family:
    </label>
    <span>
      MockJobFamily
    </span>
  </div><div
    class="info-row"
  >
    <label
      class="font-weight-bold w-25 mb-0"
    >
      Level:
    </label>
    <span>
      MockJobLevel
    </span>
  </div><div
    class="info-row border-0"
  >
    <label
      class="font-weight-bold w-25"
    >
      Description:
    </label>
    <a
      class="jdm mr-2"
      href="javascript:void(0)"
    >
      View Full Job Description
    </a>
    
    <p
      class="mb-1"
    >
       There is more than one Job Description for this job. Please refer to the 
      <a
        href="/ng/job-description-management/job-descriptions"
        target="_blank"
      >
        Job Description Management Tile
      </a>
       to view each description. 
    </p>
    
    
    
    
    
    
  </div>
</pf-company-job-detail>
`;
